ui/src/app/pages/access/access-renewal-page.tsx

```diff
@@ -52,50 +55,103 @@ const renewalStyle = {
     width: 560
   }
 };
+const isInGoodStanding = (nextReview: number): boolean => daysFromNow(nextReview) > LOOKBACK_PERIOD;
 
-export const withInvalidDateHandling = date => {
+const withInvalidDateHandling = date => {
   if (!date) {
     return 'Unavailable';
   } else {
     return displayDateWithoutHours(date);
   }
 };
 
+const computeDisplayDates = (lastConfirmedTime, bypassTime, nextReviewTime) => {
+  const userCompletedModule = !!lastConfirmedTime;
+  const userBypassedModule = !!bypassTime;
+  const lastConfirmedDate = withInvalidDateHandling(lastConfirmedTime);
+  const nextReviewDate = withInvalidDateHandling(nextReviewTime);
+  const bypassDate = withInvalidDateHandling(bypassTime);
+
+  return cond(
+    // User has bypassed module
+    [userBypassedModule, () => ({lastConfirmedDate: `${bypassDate}`, nextReviewDate: 'Unavailable (bypassed)'})],
+    // User never completed training
+    [!userCompletedModule && !userBypassedModule, () =>
+      ({lastConfirmedDate: 'Unavailable (not completed)', nextReviewDate: 'Unavailable (not completed)'})],
+    // User completed training, but is in the lookback window
+    [userCompletedModule && !isInGoodStanding(nextReviewTime), () => {
+      const daysRemaining = daysFromNow(nextReviewTime);
+      const daysRemainingDisplay = daysRemaining >= 0 ? `(${daysRemaining} day${daysRemaining !== 1 ? 's' : ''})` : '(expired)';
+      return {
+        lastConfirmedDate,
+        nextReviewDate: `${nextReviewDate} ${daysRemainingDisplay}`
+      };
+    }],
+    // User completed training and is up to date
+    [userCompletedModule && isInGoodStanding(nextReviewTime), () => {
+      const daysRemaining = daysFromNow(nextReviewTime);
+      return {lastConfirmedDate, nextReviewDate: `${nextReviewDate} (${daysRemaining} day${daysRemaining !== 1 ? 's' : ''})`};
+    }]
+  );
+};
+
+interface CompletedButtonInterface {
+  buttonText: string;
+  wasBypassed: boolean;
+  style?: React.CSSProperties;
+}
+
+const CompletedButton = ({buttonText, wasBypassed, style}: CompletedButtonInterface) => <Button disabled={true}
+    data-test-id='completed-button'
+    style={{
+      height: '1.6rem',
+      marginTop: 'auto',
+      backgroundColor: colors.success,
+      width: 'max-content',
+      cursor: 'default',
+      ...style
+    }}>
+    <ClrIcon shape='check' style={{marginRight: '0.3rem'}}/>{wasBypassed ? 'Bypassed' : buttonText}
+  </Button>;
+
+interface ActionButtonInterface {
+  isComplete: boolean;
+  wasBypassed: boolean;
+  actionButtonText: string;
+  completedButtonText: string;
+  onClick: Function;
+  disabled?: boolean;
+  style?: React.CSSProperties;
+}
+const ActionButton = (
+  {isComplete, disabled, wasBypassed, actionButtonText, completedButtonText, onClick, style}: ActionButtonInterface) => {
```

Comment:
2 nits, optional: 
`isComplete` is always set by `isInGoodStanding(getExpirationTimeFor(module))` so we could pass the module in instead
`completedButtonText` is always "Confirmed" - is this correct?  if yes, we can drop the parameter

Reply:
Sometimes the text is 'completed' - fixed
I prefer the `isComplete` property because we use this "Action / Completed / Bypassed" pattern in a few different places. This could be pulled out as a generally useful button component at a later point. Passing the module rather than the boolean makes it more specific to this component.